### DISCUSSION TITLE
New Plattform Feature : New Event after Printing or PDF creation?

### DISCUSSION THREAD

**Discussion by pri-kise** [microsoft/ALAppExtensions#14730] [General] [answered]
@JesperSchulz may I ask if there is any progress on the feature discussed in the issue #11581 ?
We would love to get access to the stream of one single record or alternativly of the correct RecVariant (Correct Filters set).

With this we could work on a more generic solution to save those PDFs in a document management system.

Right now we have to integrate in every single report to receive the correct stream and only save the document if the report isn't rendert in PreviewMode.
In the past we could you use the Event https://github.com/microsoft/AL/issues/6224

**Comment by JesperSchulz** [comment]
I'll check up on the progress of that work. I haven't heard anything back - but that doesn't mean nothing has happened :-) I will get back to you once I know more!

**Reply by JesperSchulz** [reply]
I received a reply from the developer. Unfortunately, we ran into some challenges with our test suite, which took a little more time to fix than initially anticipated. Hence the release got slightly delayed and the changes did not make the 19.0 release. We're optimistic though, that the changes will make it into the next major branch "within the next weeks" (not my words :-D). I sense a Christmas present coming your way - if all goes well... The changes should then release with the next major.

**Reply by pri-kise** [reply]
Coool, thanks for clarification. A christmas gift would be very nice :-).
Maybe you can add a comment, when this feature is merged into the main branch and available in a preview release with some little guidiance where we have to look for events to give very early feedback.

**Reply by JesperSchulz** [reply]
I'm lacking a process to do these things - notifying others... The work is done in a different team and hence I'm not getting notified proactively either. But I should hear about it at a progress review at some point. When I do, I will try to remember to update you about it :-) Wish me luck remembering that!
I'll see if I can somehow add a mechanism which reminds me of such things...

**Comment by pri-kise** [comment]
@JesperSchulz I've seen new Events in the ReportManagement for BC 20.

```al
OnAfterDocumentReady
```
```al
OnAfterDocumentDownload
```
I would like to test before the official release if these new events are useful to integrate the archiving, but there is no documentation in the object `codeunit 44 ReportManagement` to give some help with the integration.

**Reply by JesperSchulz** [reply]
The developing team promised that documentation will get updated this or next week and that we're going to post a sample app for early testing. Stay tuned just a little longer please. We're almost there.

**Reply by JesperSchulz** [reply]
The latest changes to the new events OnAfterDocumentReady and OnAfterDocumentDownload are going in over the weekend. Early doc and sample PTE are almost ready for preview too and will be made available early next week as well 🥳

**Reply by pri-kise** [reply]
Hi @JesperSchulz,
is there any progress?

**Reply by JesperSchulz** [reply]
The sample PTE and docs were slightly delayed, but will get released before Monday, I was told. The events are part of the latest insider-build off of our version 20 main branch! Feel free to take a look, and then we'll get the documentation and the sample PTE out as quickly as we can. Apologies for the delay.

**Reply by JesperSchulz** [reply]
Sample PTE is ready, but not ready to be officially shared yet. Please send me a PM on Yammer, if you'd like to get your hands on the sample extension 😊

**Comment by JesperSchulz** [comment, marked as answer]
Changes have been rolled out.
